### PR TITLE
Add movies per rightsholder

### LINKED ISSUE #9: FE ability to add movie records
## Scope

Adding movies inspiration  [workflow](https://www.figma.com/proto/saxDIb3CY3qPIBvdSiNEKt/WR-dashboard?node-id=1%3A2426&scaling=min-zoom)

- Form to capture `ImdbID`, `movie title` or `release year`
- Show a list of movie results from `themoviedb` api. Poster included
- Poster of movie is clickable for selection
- Submission button that makes a post request to graqphl to persist details
- Edit button if the user wished to change any of the fields as in workflow. Provides pre-filled form as in workflow above
- Title and Release year are readonly

Assumption:- User does not need to edit all the time and movie record is stored as is from `themoviedb`

## Gain
Ability to capture movie from right holders

## Roles
gardener: @mwaa 50 DAI
worker: @mwaa 400 DAI
reviewer: 100 DAI

### LINKED ISSUE #33: Prevent adding movies with the same IMDB id
## Scope
Prevent duplicate entries of IMDB movie records

Before movie insert:-
- Check database has no other record for IMDB ID
- Check that no other user has the same record for IMDB ID

## Gain
Enforces single entry/ownership of movie record

## Roles
gardener+worker: @mwaa 150 DAI
reviewer: 50 DAI

### PR DESCRIPTION
**Pull request by mwaa**
Please remember the [Contributing Guidelines](https://WhiteRabbitTheWorldIsYours/dashboard/blob/master/CONTRIBUTION.md) :heart:

## What does this PR do?
- Adds a search form to allow rights holder to search for movie by imdb id, title and year
- Adds an edit form to edit movie metadata
- Allow rights holder to add movie for approval

## Recommendations for testing
- Required approved user account. If not approved redirected to `/register`
- Visit `movie/add` enter movie title and year. 
- From results on table on right click movie row or select button
- Edit form is displayed. title, year, poster, review are read only
- Add to Registry confirms addition of the movie
- Display error if movie already exists in registry
- Notification on successful addition and redirect to `/titles`

## Links to relevant issues or information
Closes #9 
Closes #33

### CODE REVIEW

File: frontend/src/containers/MovieSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            <Form.Item>
+              <Button type="primary" htmlType="submit">Search</Button>
+            </Form.Item>
+          </Form>
+
+        </Col>
+        <Col className="gutter-row" xs={{ span: 24 }} lg={{ span: 18 }}>
+          <Spin spinning={loading}>
+            {
+            movie !== undefined
```
Comment (troggy) [inline frontend/src/containers/MovieSearch.tsx:117]:
```suggestion
            !movie
```

maybe?

Comment (mwaa) [reply frontend/src/containers/MovieSearch.tsx:117]: movie alone just works

File: backend/src/datasources/MovieAPI.ts [hunk 2]
```diff
@@ -11,7 +11,25 @@ class MovieAPI extends DataSource {
...
+                new FunctionExpression('attribute_not_exists', new AttributePath('sk'))
+            ]
+        };
+
+        try {
+            return await DBConnection.put({ item: movie, condition: andExpression }, {
+                condition: andExpression
+            });
+        } catch (e) {
+            if (e.message === "The conditional request failed") {
```
[the remark this answers is not in the document]
Comment (mwaa) [reply backend/src/datasources/MovieAPI.ts:28]: @troggy updated this part

**Comment by mwaa** [comment]
@troggy i have addressed the changes

- Over the year field and you should be able to see the `x` button to clear
- PK check using begins_with 
- Validation that movie id has been added and title is not undefined

File: frontend/src/containers/MovieSearch.tsx (hunk 1 above)
Comment (troggy) [reply frontend/src/containers/MovieSearch.tsx]: 😅

File: frontend/src/stores/API.ts [hunk 3]
```diff
@@ -1,20 +1,97 @@
...
-  .catch(() => ({}));
-
-export default {
-  getAuth,
-  getUser,
+// TODO: reuse this API from @whiterabbitjs/client
+import { notification } from 'antd';
+
+const config = {
+  theMovieDbApiKey: 'b1854cc7cd8f2e29da75a04a3c946e44',
```
Comment (johannbarbie) [inline frontend/src/stores/API.ts:5]: fyi: i made the repo public today :) maybe api keys should be kept outside in a config?

Comment (mwaa) [reply frontend/src/stores/API.ts:5]: yes to be handled by #50

**Pull request merged**